File: srcjs/ipywidget-ts/layers.ts
import VectorLayer from "ol/layer/Vector";
import TileLayer from "ol/layer/Tile";
import VectorTileLayer from "ol/layer/VectorTile";

import ImageLayer from "ol/layer/Image";

// WebGL
import WebGLVectorLayer from 'ol/layer/WebGLVector.js';
import WebGLTileLayer from 'ol/layer/WebGLTile.js';
import WebGLVectorTileLayer from 'ol/layer/WebGLVectorTile.js';

import VectorSource from "ol/source/Vector";

const layerCatalog: LayerCatalog = {
    TileLayer: TileLayer,
    VectorLayer: VectorLayer,
    WebGLVectorLayer: WebGLVectorLayer,
    WebGLTileLayer: WebGLTileLayer,
    VectorTileLayer: VectorTileLayer,
    WebGLVectorTileLayer: WebGLVectorTileLayer,
    ImageLayer: ImageLayer
};

// Draw interaction
const drawSource = new VectorSource({ wrapX: false });
const drawVectorLayer = new VectorLayer({
    source: drawSource
});
drawVectorLayer.set("id", "draw-features");
drawVectorLayer.set("type", "VectorLayer");

export { layerCatalog, drawSource, drawVectorLayer };

File: src/openlayers/layers.py
from __future__ import annotations

from .basemaps import BasemapLayer
from .models.layers import (
    Layer,
    LayerT,
    TileLayer,
    VectorLayer,
    WebGLTileLayer,
    WebGLVectorLayer,
    VectorTileLayer,
    WebGLVectorTileLayer
)

__all__ = [
    "TileLayer",
    "VectorLayer",
    "WebGLTileLayer",
    "WebGLVectorLayer",
    "BasemapLayer",
    "VectorTileLayer",
    "WebGLVectorTileLayer"
]
